### PR TITLE
feat: added a form for creating tokens

### LINKED ISSUE #3562: [Feature] Use generate button pressing enter in dialog for creating a token
## Recommending Features

image

When creating a token, you need to set a name for this token. It would be great if you confirm the name with pressing "Enter" and execute the "Generate" button.

### PR DESCRIPTION
**Pull request by asheerrizvi**
## Change Summary

Changed generate token to a form to make it more accessible for keyboard users.

Closes #3562 

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Not required.

## Additional information / screenshots (optional)

Not required. 

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui/components/tabs/auth/ApiTokenManagement.vue [hunk 1]
```diff
@@ -106,17 +106,29 @@ onMounted(() => {
...
+        no-style
+        autocomplete="off"
+        @finish="generateToken"
+      >
         <a-input v-model:value="selectedTokenData.description" :placeholder="$t('labels.description')" />
+
         <!-- Generate -->
         <div class="flex flex-row justify-center">
-          <a-button type="primary" @click="generateToken"> {{ $t('general.generate') }} </a-button>
+          <button type="submit" class="ant-btn ant-btn-primary">
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/tabs/auth/ApiTokenManagement.vue:127]: Why use `button` and apply `ant-btn` and not just `a-button`? 😄 
if you need to add `type="submit"` you can pass it as `html-type` prop, i.e. `html-type="submit"`

```html
<a-button html-type="submit" type="primary">
   {{ $t('general.generate') }}
</a-button>
```

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

File: packages/nc-gui/components/tabs/auth/ApiTokenManagement.vue (hunk 1 above)
Comment (asheerrizvi) [reply packages/nc-gui/components/tabs/auth/ApiTokenManagement.vue:127]: That's a dummy mistake lol. I'll do the fix.

Comment (asheerrizvi) [reply packages/nc-gui/components/tabs/auth/ApiTokenManagement.vue]: @bcakmakoglu done!

**Review by bcakmakoglu — APPROVED** [review]

**Comment by o1lab** [comment]
@all-contributors please add @asheerrizvi  for code

**Pull request merged**